angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/widget-card.html

```diff
@@ -107,7 +109,7 @@
             <portlet-icon></portlet-icon>
           </div>
         </a>
-        <button aria-labelledby="goToApps-{{portlet.nodeId}} appTitle_portlet.title-{{portlet.nodeId}}" class="launch-app-button"  ng-click="widgetCtrl.maxStaticPortlet(portlet)">
+        <button aria-label="{{ portlet.widgetConfig.launchText ? portlet.widgetConfig.launchText : 'Launch full app' }}" class="launch-app-button"  ng-click="widgetCtrl.maxStaticPortlet(portlet)">
```

Comment:
I think there (should be) different guidance for sighted vs screen-reader-facing text. 

For sighted, "Don't repeat yourself" is decently good advice - the app card's title already visually associates with the launch button, so the launch button should observe the Thumperian Principle and say as little as possible if it has nothing interesting to say.

Contrastingly, non-sighted users will have difficulty visually associating the card title with the launch button on that card -- at best one would have to remember what card context you're in between the two elements; at worst it'll just be an out-of-context generic "click here" style link. So for the `aria-label` being as specific as possible, even at the risk of repetition of what's already in the card title, is the way to go.